src/script/lua_api/l_item.cpp

```diff
@@ -437,8 +437,32 @@ int LuaItemStack::l_peek_item(lua_State *L)
 int LuaItemStack::l_equals(lua_State *L)
 {
 	LuaItemStack *o1 = checkobject(L, 1);
-	ItemStack &item1 = o1->m_stack;
+
+	void *p = lua_touserdata(L, 2);
+	if (p == NULL) {
+		lua_pushboolean(L, false);
+		return 1;
+	}
+
+	if (!lua_getmetatable(L, 2)) {
+		lua_pushboolean(L, false);
+		return 1;
+	}
+
+	lua_getfield(L, LUA_REGISTRYINDEX, className);
+	if (!lua_rawequal(L, -1, -2)) {
+		lua_pushboolean(L, false);
+		return 1;
+	}
+
 	LuaItemStack *o2 = checkobject(L, 2);
+
+	if (o1 == NULL || o2 == NULL) {
```

Comment:
checkobject never returns NULL

Reply:
right, that's a leftover from before i figured out how to do this correctly, will remove.